exporters/ostream/src/metric_exporter.cc

```diff
@@ -79,13 +79,33 @@ sdk::common::ExportResult OStreamMetricExporter::Export(
 
   for (auto &record : data.instrumentation_info_metric_data_)
   {
-    printInstrumentationInfoMetricData(record);
+    printInstrumentationInfoMetricData(record, data);
```

Comment:
Can we call `printResources` here and keep `printInstrumentationInfoMetricData` unchanged. Something like - 

```cpp
for (auto &record : data.instrumentation_info_metric_data_)
  {
    printResource(*data.resource_)
    printInstrumentationInfoMetricData(record);
}
```

Reply:
Also, ostream tests need to be updated too.

Reply:
I updated the tests, but factoring out the `printResource` function would be a lot more work, since `printInstrumentationInfoMetricData` currently requires a lock on `sout_`. Currently, the `trace_sdk` has the concept of a [Recordable interface](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/exporters/ostream/src/span_exporter.cc#L47-L96) that is used for encapsulating all recordable data under one abstraction. We'd need something similar on the metrics side to `printRecordable(recordable)` I think.

Reply:
yes agree, I see the issue with proper encapsulation. One minor suggestion, should we rename `printInstrumentationInfoMetricData` method to
`printMetricData(record, resource&) `?

Reply:
If it's alright, I'd like to leave the names as is to reduce the footprint of merge conflicts with #1507.

Reply:
yes good point. Thanks.